warp10/src/main/java/io/warp10/script/functions/SHRINK.java

```diff
@@ -49,13 +49,15 @@ public Object applyOnElement(Object element) {
         if (element instanceof GeoTimeSerie) {
           GeoTimeSerie gts = (GeoTimeSerie) element;
 
-          if (GTSHelper.nvalues(gts) > Math.abs(shrinkto)) {
-            if (shrinkto < 0) {
-              GTSHelper.sort(gts, true);
-            } else {
-              GTSHelper.sort(gts, false);
+          if (shrinkto != Long.MIN_VALUE && shrinkto != Long.MAX_VALUE) {
```

Comment:
don't need to check that shrinkto != Long.Max_Value

Reply:
It saves a cycle. There is no need to call shrink in this case.

Reply:
Then you can also save a cycle by not calling shrink when shrinkto is equal to Min_Value + 1 (since it is equal to abs(MAX_VALUE)). Modify the condition accordingly.

Reply:
Code updated with Integer limits, for a safe cast line 58.